### ISSUE TITLE
action_set uses dot notation for nesting

### ISSUE THREAD

**Issue by stub42** [canonical/operator#219]
ActionEvent.set_results's argument is a dictionary, but the keys are in juju dot-notation to express nesting. So instead of passing a standard Python data structure like ```{'a': {'b': 1, 'c': 2}}```, instead the charm author needs to convert the nested structure it to the non-standard, Juju specific ```{'a.b': 1, 'a.c': 2}``` notation.

Instead, I think ActionEvent should handle serializing the results into Juju notation suitable for the action-set tool. It should also handle quoting, or raise meaningful exceptions if the keys or values not acceptable (no . or _ in keys etc.). Long term, it may be better for action-set to accept a yaml or json data structure to avoid custom serialization mechanisms, originally designed for use by bash charms IIRC.

**Comment by jameinel** [comment]
Agreed. We should accept nested dicts and turn them into dotted notation.
